### ISSUE TITLE
Model save issues with custom optimizer - DPKerasSGDOptimizer

### ISSUE THREAD

**Issue by avssridhar** [keras-team/keras#17436]
****Problem**** 
A keras sequential Model was compiled using DPKerasSGD optimizer and saved. But when the saved model is loaded it throws the below error

"Unknown optimizer: 'DPOptimizerClass'. Please ensure you are using a `keras.utils.custom_object_scope` and that this object is included in the scope. See https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details"

**Expected Behaviour**
Since the model was already compiled using the optimizer it should run perfectly fine when fitted with the data. 

Whereas the same works when the optimizer is replaced. Please provide a solution on how to save the model with this custom optimizer.

**Standalone code to reproduce the issue**

#building the model
model = tf.keras.Sequential([
    tf.keras.layers.Dense(128, activation='relu'),
    tf.keras.layers.Dense(256, activation='relu'),
    tf.keras.layers.Dense(256, activation='relu'),
    tf.keras.layers.Dense(1, activation='sigmoid')])

#compiling the model
  model.compile(
    loss=tf.keras.losses.CategoricalCrossentropy(from_logits=True,reduction=tf.losses.Reduction.NONE),
    optimizer=tensorflow_privacy.DPKerasSGDOptimizer(l2_norm_clip=1,noise_multiplier=2,learning_rate=0.01),
    metrics=[
        tf.keras.metrics.BinaryAccuracy(name='accuracy'),
        tf.keras.metrics.Precision(name='precision'),
        tf.keras.metrics.Recall(name='recall')
    ])

#saving the model
model.save('my_model')

#loading the model
keras.models.load_model('mymodel')

#throws the error

**Comment by sushreebarsa** [comment]
@avssridhar 
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thank you!
